### PR TITLE
Fix loads the entire frame on the iframe. ⚡

### PR DESCRIPTION
**Pull request by Nishitbaria**
## Fixes Issue
#1612 

Closes #issue_number
#1612 

## Changes proposed

- Added a state variable `showIframe` to track whether the iframe should be rendered or not.
- Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
- Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
- Update

## Screenshots

https://github.com/Spyware007/Animating-Buttons/assets/85815172/6592efbd-5fe3-47c1-a9c1-1a3c45c9679c

## Checklist

- [x] You've linked this PR to the correct issue.
- [x] You have checked that the code is working correctly.
- [x] You ⭐️ the repository!

### CODE REVIEW

**Comment by Nishitbaria** [comment]
hey @Spyware007 review PR

**Comment by adity2911** [comment]
I think on click, the iframe disappears?

**Comment by Nishitbaria** [comment]
@adity2911  please send me video of your problem because i am unable to see issue of your problem?

**Comment by adity2911** [comment]
You will have to record a video and show it to me. I cannot view buttonsCard on the deployement page.

**Comment by Nishitbaria** [comment]
hey @adity2911   You are not able to see button cards because your firebase quota is reached you can check this screenshot,by going to website,On my local I have created buttons and then showing you demo,you can your main website also

**Comment by adity2911** [comment]
image
Not working still.

**Comment by Nishitbaria** [comment]
check this on your main website @adity2911 


WhatsApp Image 2023-06-29 at 00 24 58

**Comment by adity2911** [comment]
I can use my firebase alright.
I tried to run the code on my local machine and it failed there.

**Comment by Nishitbaria** [comment]
@adity2911 Make Sure that you have enable github Login in firebase Authentication Section with 
Client Id and Client Secret 
and also share the video which will be better so we can inspect the issue in best way

**Comment by adity2911** [comment]
It is the same issue as what i had mentioned on my issues tab.
check the video i uploaded there once again. It's the same issue now also.

**Comment by Nishitbaria** [comment]
https://animating-buttons-silk.vercel.app/
@adity2911 Kindly check my deployment link I have hosted from vercel and I think you haven't add you domain in firebase Authorized domain section as mentioned below so kindly check upon it

until that you can check my deployment link

image

**Comment by Nishitbaria** [comment]
You can also check this video for reference
https://github.com/Spyware007/Animating-Buttons/assets/85815172/2b2d7f8c-8bfe-42a6-bb70-e7bb673b9f69

**Comment by adity2911** [comment]
If you look closely at the video, you might notice that the issue was not for all buttons. There are some particular buttons that are facing this issue. Also as i have said, I tried to run your code on my localHost and it was not working there aslo. Plus the code broke the download functionality also.

**Comment by Nishitbaria** [comment]
@adity2911  Bro for running my code you need to setup github auth in firebase and add   your url to authorized domains ,then run it on local and you can check my deployed url it contains buttons and tell me which buttons are creating   ,you can check my deployed url by adding buttons ,there is no problem in my  website   ,please merge it And FIRST OF ALL FIX YOUR MAIN WEBSITE ,because it has firebase quota reached  and no user can now see your buttons.

**Comment by Souravvmishra** [comment]
@Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?

**Comment by Nishitbaria** [comment]
> @Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?


- Added a state variable `showIframe` to track whether the iframe should be rendered or not.
- Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
- Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
- Updated testimonial js

**Comment by Nishitbaria** [comment]
> @Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?


- Added a state variable `showIframe` to track whether the iframe should be rendered or not.
- Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
- Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
- Updated testimonial js

**Comment by adity2911** [comment]
I checked the link you provided.
The download button is now broken.
what did you exactly fix?

**Comment by Nishitbaria** [comment]
> I checked the link you provided. The download button is now broken. what did you exactly fix?

wait a minute

**Comment by Souravvmishra** [comment]
> > @Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?
> 
> 
> - Added a state variable `showIframe` to track whether the iframe should be rendered or not.
> - Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
> - Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
> - Updated testimonial js         

How Does This Fix The Issue?

**Comment by Nishitbaria** [comment]
> > > @Nishitbaria Can You Explain The Logic, What Changes You Made In Order To Fix The Bug?
> > 
> > 
> > 
> > * Added a state variable `showIframe` to track whether the iframe should be rendered or not.
> > * Modified the `handleButtonClick` function to toggle the value of `showIframe` when the button is clicked.
> > * Wrapped the `iframe` element with a conditional statement to render it only when `showIframe` is `true`.
> > * Updated testimonial js
> 
> How Does This Fix The Issue?

The issue described in video by @adity2911 is now resolved, that what I have fixed you can now.  : https://animating-buttons-silk.vercel.app/

File: src/components/common/Card/Card.jsx [hunk 1]
```diff
@@ -47,20 +48,25 @@ export default function Card({ button }) {
     };
     fetchUser();
   }, [user]);
+  const handleButtonClick = () => {
```
Comment (adity2911) [inline src/components/common/Card/Card.jsx:51]: the function is added but not called anywhere.

**Review by adity2911 — CHANGES_REQUESTED** [review]

**Comment by Souravvmishra** [comment]
@Nishitbaria Can You Explain How Adding A State Variable Fixed The Issue? & if I'm Getting you wrong what else you did that fixed the issue.

File: src/components/common/Card/Card.jsx (hunk 1 above)
Comment (Nishitbaria) [reply src/components/common/Card/Card.jsx:51]: It is not need to be called ,That i have kept just for debuggging

**Comment by Nishitbaria** [comment]
> @Nishitbaria Can You Explain How Adding A State Variable Fixed The Issue? & if I'm Getting you wrong what else you did that fixed the issue.



> @Nishitbaria Can You Explain How Adding A State Variable Fixed The Issue? & if I'm Getting you wrong what else you did that fixed the issue.

Main thing is that i have solved this issue ,it is working fine as per your requirements ,you can merge my pull request

**Comment by Souravvmishra** [comment]
& Just for the reference purpose, even if we remove the changes you have done, it will still work fine for your buttons. You can check it out yourself.

**Comment by Nishitbaria** [comment]
> > > @Nishitbaria Can You Explain How Adding A State Variable Fixed The Issue? & if I'm Getting you wrong what else you did that fixed the issue.
> 
> > Main thing is that i have solved this issue ,it is working fine as per your requirements ,you can merge my pull request
> 
> Sorry Brother, We're not satisfied with your Solution yet

what are you requirement,what you want me to do(As I have resolved issue mentioned by @adity2911 )

**Comment by Nishitbaria** [comment]
> & Just for the reference purpose, even if we remove the changes you have done, it will still work fine for your buttons. You can check it out yourself.

Issues are meant to be created and assigned when they are present and when somebody is assigned ,he analyzes your code,add his improvements ,but you assigned me issue,i worked on it and now you are not merging .due to your mistake.
Why you assigned this issue me,didn't you checked this when you assign me this issue ,I have worked on it actively after that,and now you are telling ,you dont need my changes ,when a issue is assigned ,it should be checked and then assigned,that your mistake .

`please merge my pull request ,please dont let  our invested time be wasted 🙏`

**Comment by Nishitbaria** [comment]
@adity2911 @Souravvmishra

**Comment by Souravvmishra** [comment]
@adity2911 We never refused to merge your code. We'll surely do that once we are satisfied with your Solution.

All you have to explain is how is your solution fixing the issue?
What was the purpose of adding the state variable & that conditional rendering? 
How are all these things fixing the issue?

**Comment by adity2911** [comment]
> > & Just for the reference purpose, even if we remove the changes you have done, it will still work fine for your buttons. You can check it out yourself.
> 
> Issues are meant to be created and assigned when they are present and when somebody is assigned ,he analyzes your code,add his improvements ,but you assigned me issue,i worked on it and now you are not merging .due to your mistake. Why you assigned this issue me,didn't you checked this when you assign me this issue ,I have worked on it actively after that,and now you are telling ,you dont need my changes ,when a issue is assigned ,it should be checked and then assigned,that your mistake .
> 
> `please merge my pull request ,please dont let our invested time be wasted 🙏`

Adding two lines of code that doesnot do anything is not considered to be a investment of time. It's a waste of time.

**Comment by adity2911** [comment]
> > @Nishitbaria We never refused to merge your code. We'll surely do that once we are satisfied with your Solution.
> > All you have to explain is how is your solution fixing the issue? What was the purpose of adding the state variable & that conditional rendering? How are all these things fixing the issue?
> 
> I implemented state variable ,which iframe variable holds the state of iframe and doesn't allow it to re-render until handlebuttonclick is called and i have just defines that function ,for your future use if you want to use in any other case ,i can also remove it .My logic was simple that iframe a state variable,preserves the state and does not change in any case. @Souravvmishra

Can you explain when the state of the variable is cahanged? I dont seem to find any instance where the state is changed.

**Comment by Nishitbaria** [comment]
> > > & Just for the reference purpose, even if we remove the changes you have done, it will still work fine for your buttons. You can check it out yourself.
> > 
> > 
> > Issues are meant to be created and assigned when they are present and when somebody is assigned ,he analyzes your code,add his improvements ,but you assigned me issue,i worked on it and now you are not merging .due to your mistake. Why you assigned this issue me,didn't you checked this when you assign me this issue ,I have worked on it actively after that,and now you are telling ,you dont need my changes ,when a issue is assigned ,it should be checked and then assigned,that your mistake .
> > `please merge my pull request ,please dont let our invested time be wasted 🙏`
> 
> Adding two lines of code that doesnot do anything is not considered to be a investment of time. It's a waste of time.me

Your issue was resolved in my 2 lines of code ,Why you assigned  me this issue  ,you should have think that as a maintainer ans this is not good ,i have explain my logic in previous #pl 

> > > @Nishitbaria We never refused to merge your code. We'll surely do that once we are satisfied with your Solution.
> > > All you have to explain is how is your solution fixing the issue? What was the purpose of adding the state variable & that conditional rendering? How are all these things fixing the issue?
> > 
> > 
> > I implemented state variable ,which iframe variable holds the state of iframe and doesn't allow it to re-render until handlebuttonclick is called and i have just defines that function ,for your future use if you want to use in any other case ,i can also remove it .My logic was simple that iframe a state variable,preserves the state and does not change in any case. @Souravvmishra
> 
> Can you explain when the state of the variable is cahanged? I dont seem to find any instance where the state is changed.

You issue requires that state of iframe does not change,because if it changes then iframe will be re-rendered.so i have fixed the state variable.

**Comment by Spyware007** [comment]
@Nishitbaria @adity2911 @Souravvmishra 
Please Stop!
@Nishitbaria Sorry for the inconvenience but whatever you had implemented is not a desired solution! 

Closing this PR!

The simplest solution to this is to remove the hyperlink as per my knowledge for those buttons where anchor tag is used instead of button tag!

This can't be solved by making changes in the code, have to change the HTML code from the backend!

**Comment by Nishitbaria** [comment]
> @Nishitbaria @adity2911 @Souravvmishra Please Stop! @Nishitbaria Sorry for the inconvenience but whatever you had implemented is not a desired solution!
> 
> Closing this PR!
> 
> The simplest solution to this is to remove the hyperlink as per my knowledge for those buttons where anchor tag is used instead of button tag!
> 
> This can't be solved by making changes in the code, have to change the HTML code from the backend!

Then why you assign issue when it is resolved already ,thats the problem.

**Comment by adity2911** [comment]
I did not assign you. I asked you to look if you can solve it or not.

**Comment by Nishitbaria** [comment]
> I did not assign you. I asked you to look if you can solve it or not.

ok

**Pull request closed without merging**